versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/SetMojo.java

```diff
@@ -348,9 +350,22 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                     Pattern.compile(RegexUtils.convertWildcardsToRegex(fixNullOrEmpty(oldVersion, "*"), true));
 
             for (Model m : reactor.values()) {
-                final String mGroupId = PomHelper.getGroupId(m);
-                final String mArtifactId = PomHelper.getArtifactId(m);
-                final String mVersion = PomHelper.getVersion(m);
+                Map<String, String> properties = ofNullable(m.getProperties())
+                        .map(p -> p.entrySet().parallelStream()
+                                .collect(Collectors.toMap(e -> e.getKey().toString(), e -> e.getValue()
+                                        .toString())))
```

Comment:
From: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-

> The returned Collector is not concurrent. For parallel stream pipelines, the combiner function operates by merging the keys from one map into another, which can be an expensive operation. If it is not required that results are inserted into the Map in encounter order, using [toConcurrentMap(Function, Function)](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toConcurrentMap-java.util.function.Function-java.util.function.Function-) may offer better parallel performance.

So for such operation where not many elements are to compute we can use simple stream instead of parallel Stream.

I hope that we will have a less problems with simple stream.

Reply:
Done.